tests/components/mhz19/test_sensor.py

```diff
@@ -101,25 +101,24 @@ async def test_co2_sensor(mock_function, hass):
 async def test_temperature_sensor(mock_function, hass):
     """Test temperature sensor."""
     client = mhz19.MHZClient(co2sensor, "test.serial")
-    sensor = mhz19.MHZ19Sensor(client, mhz19.SENSOR_TEMPERATURE, None, "name")
+    sensor = mhz19.MHZ19Sensor(client, mhz19.SENSOR_TEMPERATURE, "name")
     sensor.hass = hass
     sensor.update()
 
     assert sensor.name == "name: Temperature"
     assert sensor.state == 24
-    assert sensor.unit_of_measurement == TEMP_CELSIUS
+    assert sensor.native_unit_of_measurement == TEMP_CELSIUS
     assert sensor.should_poll
     assert sensor.extra_state_attributes == {"co2_concentration": 1000}
 
 
 @patch("pmsensor.co2sensor.read_mh_z19_with_temperature", return_value=(1000, 24))
 async def test_temperature_sensor_f(mock_function, hass):
     """Test temperature sensor."""
-    client = mhz19.MHZClient(co2sensor, "test.serial")
-    sensor = mhz19.MHZ19Sensor(
-        client, mhz19.SENSOR_TEMPERATURE, TEMP_FAHRENHEIT, "name"
-    )
-    sensor.hass = hass
-    sensor.update()
+    with patch.object(hass.config.units, "temperature_unit", TEMP_FAHRENHEIT):
+        client = mhz19.MHZClient(co2sensor, "test.serial")
+        sensor = mhz19.MHZ19Sensor(client, mhz19.SENSOR_TEMPERATURE, "name")
+        sensor.hass = hass
+        sensor.update()
 
-    assert sensor.state == 75.2
+        assert sensor.state == "75"
```

Comment:
Wondering why the value is converted to a string though. I don't think there is a need for it.
https://github.com/home-assistant/core/blob/6cf312f3c8dcefe86d83b78a0670203393f40ee1/homeassistant/components/sensor/__init__.py#L292-L294

Reply:
The conversion to string here is a bit redundant as the state will anyway be converted to a string before being written to the state machine by `_async_write_ha_state` in the `Entity` base class.

Reply:
I think a separate PR getting rid of the string conversion would be welcome 👍